src/ring_theory/ideal/nc_jacobson_radical.lean

```diff
@@ -0,0 +1,251 @@
+/-
+Copyright (c) 2022 Haruhisa Enomoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Haruhisa Enomoto
+-/
+import ring_theory.ideal.basic
+import tactic.noncomm_ring
+/-!
+# Noncommutative Jacobson radical
+
+In this file, we define the Jacobson radical of a ring and prove its basic properties.
+
+Let `R` be a (possibly noncommutative) ring. The *Jacobson radical* of `R` is defined
+to be the intersection of all maximal *left* ideals of `R`, which coincides with that of
+all maximal *right* ideals, so it is a two-sided ideal.
+
+## Main definitions
+
+* `nc_jacobson R : ideal R`: the Jacobson radical of a ring `R`, that is,
+  the intersection of all maximal *left* ideals.
+
+## Main statements
+
+* `mem_nc_jacobson_tfae`: some equivalent conditions for an element in a ring
+  to be contained in the Jacobson radical.
+
+* `nc_jacobson_symm`: the definition of the Jacobson radical is left-right symmetric, that is,
+  the intersection of all maximal left ideals coincides with that of all maximal right ideals.
+-/
+
+universe u
+open mul_opposite
+
+section monoid
+variables {M : Type u} {a : M} {b : M} [monoid M]
+/-! ### Some lemmas on inverses in monoids -/
+
+lemma is_unit.op (h : is_unit a) : is_unit (op a) :=
+let ⟨⟨_, _, hab, hba⟩, rfl⟩ := h in ⟨⟨_, _, by rw [←op_mul, ←op_one, hba],
+  by rw [←op_mul, ←op_one, hab]⟩, rfl⟩
+
+lemma is_unit.unop {x : Mᵐᵒᵖ} (h : is_unit x) : is_unit (unop x) :=
+let ⟨⟨_, _, hxa, hax⟩, rfl⟩ := h in ⟨⟨_, _, by rw [←unop_mul, ←unop_one, hax],
+  by rw [←unop_mul, ←unop_one, hxa]⟩, rfl⟩
+
+lemma is_unit_op_iff_is_unit {a : M} : is_unit (op a) ↔ is_unit a :=
+⟨λ h, unop_op a ▸ h.unop, is_unit.op⟩
+
+lemma is_unit_unop_iff_is_unit {x : Mᵐᵒᵖ} : is_unit (unop x) ↔ is_unit x :=
+⟨λ h, op_unop x ▸ h.op, is_unit.unop⟩
```

Comment:
It might be possible to golf these a little using [`units.op_equiv`](https://leanprover-community.github.io/mathlib_docs/algebra/group/opposite.html#units.op_equiv)

Reply:
Thanks! I rewrote the first and second using it. I think the rest are short and simple enough.